### PR TITLE
Removing custom DB logic in favour of flask-sqlalchemy

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
I have removed most of our Database connection and session handling logic as that is all done by `flask-sqlalchemy` which we are already importing as a dependency.
This also ties the database better into the request context.

### How to review 
Deploy to ElasticBeanstalk
Resize Database
Validate connections come back after resize has finished

Fixes https://github.com/ONSdigital/eq-survey-runner/issues/1376

### CODE REVIEW

File: app/data_model/database.py [hunk 1]
```diff
@@ -62,64 +57,34 @@ def __repr__(self):
 
 class Database:
 
-    def __init__(self, driver, database_name, host=None, port=None,
-                 username=None, password=None, setup_attempts=SETUP_ATTEMPTS,
-                 setup_retry_delay=SETUP_RETRY_DELAY_SECONDS):
```
Comment (MIslam92) [inline app/data_model/database.py]: Any reason the all the setup_retry_delay stuff was removed? When your initialising the db etc. How does SQLAlchemy with the retry during setup if a error occurs itself?

File: tests/app/authentication/test_database_session_manager.py [hunk 2]
```diff
@@ -16,10 +16,9 @@ class TestSessionManager(AppContextTestCase):
 
     def setUp(self):
         super().setUp()
-        self.database = Mock(Database(driver='sqlite',
-                                      database_name='',
-                                      setup_attempts=1,
-                                      setup_retry_delay=0))
+        self.database = Mock(Database(application=self._app,
+                                      driver='sqlite',
+                                      database_name=''))
```
Comment (MIslam92) [inline tests/app/authentication/test_database_session_manager.py]: Should this not have some name something like test_db?

File: app/data_model/database.py (hunk 1 above)
Comment (jonnyshaw89) [reply app/data_model/database.py]: It kept retrying when i tested but you should test too.

File: tests/app/authentication/test_database_session_manager.py (hunk 2 above)
Comment (jonnyshaw89) [reply tests/app/authentication/test_database_session_manager.py]: Its mocked so it makes no difference

**Review by MIslam92 — APPROVED** [review]

**Comment by ajmaddaford** [comment]
I think we should look to move the database setup code out of database.py and into setup.py as:
- It doesn't make sense for the database connection code to sit with the models (we might want to consider renaming database.py to models.py)
- It would simplify the code to check the database state on startup in https://github.com/ONSdigital/eq-survey-runner/pull/1347

**Comment by jonnyshaw89** [comment]
@ajmaddaford Ready for you to review again

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**